custom_model_runner/datarobot_drum/drum/language_predictors/r_predictor/r_predictor.py

```diff
@@ -102,6 +102,59 @@ def _get_sparse_colnames(kwargs):
             sparse_colnames = ro.rinterface.NULL
         return sparse_colnames
 
+    def _replace_sanitized_class_names(self, data):
+        """ Match prediction data labels to project class labels.
+        Note that this contains only logic specific to R name
+        sanitization and relies on marshal_predictions() for
+        language neutral cases like matching floats and bools """
+
+        # get class labels
+        if (
+            self._r_positive_class_label is not ro.rinterface.NULL
+            and self._r_negative_class_label is not ro.rinterface.NULL
+        ):
+            class_labels = [self._r_negative_class_label, self._r_positive_class_label]
+        elif self._r_class_labels is not None:
+            class_labels = self._r_class_labels
+        else:
+            raise DrumCommonException("Class labels not available for classification.")
+
+        prediction_labels = data.columns
+
+        # if the labels match then do nothing
+        if set(prediction_labels) == set(class_labels):
+            return data
+
+        # check for match after make.names is applied to class labels
+        sanitized_class_labels = ro.r["make.names"](class_labels)
+        if set(prediction_labels) == set(sanitized_class_labels):
+            if len(set(sanitized_class_labels)) != len(sanitized_class_labels):
+                raise DrumCommonException("Class label names are ambiguous.")
+            label_map = dict(zip(sanitized_class_labels, class_labels))
+            # return class labels in the same order as prediction labels
+            ordered_labels = [label_map[l] for l in prediction_labels]
+            data.columns = ordered_labels
+            return data
+
+        def floatify(f):
+            try:
+                return float(f)
+            except ValueError:
+                return f
+
+        # check for match after sanitized float strings (e.g. X7.1) are converted to plain floats
```

Comment:
Why would they be prefixed with X?

Reply:
Some R estimators (like `gbm`) use `make.names` for dataframe colnames, so:
```
> make.names(c('a-b', '7.1', '7', 7.1, 7))
[1] "a.b"  "X7.1" "X7"   "X7.1" "X7"  
```

Reply:
so your code wouldn't properly deal with class labels where some labels are floats right... That's probably fine

Reply:
The code above it should handle mixed float and non-float classes if that's what you are asking.  I'm guessing mixed classes will always be strings and not run into the issue where pred labels are '7.0' and class labels are '7'.